lib/filesystem/FileStream.cpp

```diff
@@ -0,0 +1,148 @@
+#include "StdInc.h"
+#include "FileStream.h"
+#include "../minizip/unzip.h"
+
+#include <cstdio>
+
+
+#ifdef _WIN32
+	#ifndef _CRT_SECURE_NO_WARNINGS
+		#define _CRT_SECURE_NO_WARNINGS
+	#endif
+	#include <cwchar>
+	#define CHAR_LITERAL(s) L##s
+	#define DO_OPEN(name, mode) _wfopen(name, mode)
+	using CharType = wchar_t;
+#else
+	#define CHAR_LITERAL(s) s
+	#define DO_OPEN(name, mode) fopen(name, mode)
+	using CharType = char;
+#endif
+
+#define GETFILE static_cast<std::FILE*>(filePtr)
+
+voidpf ZCALLBACK MinizipOpenFunc(voidpf opaque, const void* filename, int mode)
+{
+    const CharType* mode_fopen = [mode]() -> const CharType*
```

Comment:
I usually assign variables a value at initialization and never change it, whenever possible. It's just a habit of mine. It does not have to be a lambda here.